### PR TITLE
v3 (BREAKING CHANGES)

### PR DESCRIPTION
**Pull request by tgharold**
Major rewrite to move to using the `IFeatureManagerSnapshot` and `ISessionManager` interfaces from [Microsoft.FeatureManagement](https://www.nuget.org/packages/Microsoft.FeatureManagement/).  This will allow layering of `ISessionManager` implementations to give flexibility in how the feature value is calculated.

There's opposing viewpoints on this that say we could lean into `FeatureFlagDefinition` and `FeatureFilterConfiguration` for deciding whether a feature should be turned on/off if it was not already set via the `ISessionManager`.  We're deferring those choices until later; even if we do go down that route it will (likely) be a minor change compared to what we had to do for v3.

see: CHANGE_LOG.md file for full details

### CODE REVIEW

**Comment by tgharold** [comment]
# Current UI

Screen Shot 2022-06-20 at 9 36 48 AM

## Notes

- May need language at the top explaining which database table is being updated by these rows?
- Null/False/True may need more explanation.
- We've discussed a 3-state slider / button panel thingy.

The "null" means that the feature is not set in the database.  It might be defined at a layer above/below this point.  We may need RimDev.FeatureFlags to be better about supplying information about how the feature value is calculated (it's a priority layered system, the first layer to give a definitive true/false answer is what gets returned).

## Links

https://stackoverflow.com/a/57424477
https://stackoverflow.com/a/56790599
https://codepen.io/ahaywood/pen/yLJbRbP?editors=1100

**Comment by tgharold** [comment]
# Platform UI Styling

image

**Comment by tgharold** [comment]
# Updated UI

image

File: samples/FeatureFlags.AspNetCore/Startup.cs [hunk 1]
```diff
@@ -36,13 +60,15 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
                 app.UseDeveloperExceptionPage();
             }
 
-            app.UseFeatureFlags(options);
-            //app.UseFeatureFlagsUI(options);
+            app.UseRimDevFeatureFlags();
+            app.UseRimDevFeatureFlagsUi();
```
Comment (billbogaiv) [inline samples/FeatureFlags.AspNetCore/Startup.cs:64]: Do we want to keep the `Ui`-part as-is or change to `UI` to mirror casing with `MapFeatureFlagsUI` (or vice-versa)?

**Review by billbogaiv — COMMENTED** [review]
Looks good—pretty excited to get this into our apps... been a long time coming 😄

**Review by billbogaiv — APPROVED** [review]

**Pull request merged**